EventsExpress/ClientApp/src/components/event/filter/parts/location/location-filter.js

```diff
@@ -0,0 +1,113 @@
+import { FilterExpansionPanel } from "../../expansion-panel/filter-expansion-panel";
+import { change, getFormValues } from "redux-form";
+import React from "react";
+import { connect } from "react-redux";
+import { enumLocationType } from "../../../../../constants/EventLocationType";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormControl from "@material-ui/core/FormControl";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import Radio from "@material-ui/core/Radio";
+import { Field } from "redux-form";
+import { LocationMapWithCircle } from "../../../../helpers/form-helpers/location";
+import DisplayMap from "../../../../event/map/display-map";
+import "../../../slider.css";
+const LocationFilter = ({ dispatch, location, formValues }) => {
+  const clear = () => dispatch(change("filter-form", "location", {type: null} ));
+
+  const onChangeLocationType = (event) => {
+    const type = Number(event.target.value);
+    if (type === enumLocationType.map) {
+      dispatch(
```

Comment:
Please use mapDispatchToProps and do not call dispatch inside components directly.

Reply:
Done